### PR TITLE
feat(database): Aurora serverless support

### LINKED ISSUE #33: support aurora serverless
Let's support aurora serverless for the database

### PR DESCRIPTION
**Pull request by pahud**
- support opt-in for aurora serverless

- [x] integration testing in `ap-northeast-1`
- [x] integration testing in `cn-northwest-1`

(please note: aurora serverless is not supported in `cn-north-1`)

Fixes #33

### CODE REVIEW

**Comment by pahud** [comment]
copy @wchaws

**Comment by pahud** [comment]
Interesting `cn-northwest-1` actually supports aurora serverless but not listed in https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.AuroraFeaturesRegionsDBEngines.grids.html

Let's create a `AURORA_SERVERLESS_SUPPORTED_REGION` array to hold all supported regions and throw Error when deploy to unsupported regions.

File: src/keycloak.ts [hunk 1]
```diff
@@ -124,20 +162,59 @@ export interface DatabaseProps {
...
+      // allow from the whole vpc cidr
+      this.dbinstance.connections.allowFrom(ec2.Peer.ipv4(props.vpc.vpcCidrBlock), ec2.Port.tcp(this._mysqlListenerPort));
+      this.clusterEndpointHostname = this.dbinstance.dbInstanceEndpointAddress;
+      this.clusterIdentifier = this.dbinstance.instanceIdentifier;
+      this.connections = this.dbinstance.connections;
+      printOutput(this, 'DBSecretArn', this.dbinstance.secret!.secretArn);
+    };
+
+    printOutput(this, 'clusterEndpointHostname', this.clusterEndpointHostname);
+    printOutput(this, 'clusterIdentifier', this.clusterIdentifier);
```
Comment (wchaws) [inline src/keycloak.ts:214]: Most of the code are the same @pahud, can we refactor it?

File: src/keycloak.ts [hunk 2]
```diff
@@ -8,6 +8,28 @@ import * as rds from '@aws-cdk/aws-rds';
...
+  'ap-south-1',
+  'ap-northeast-1',
+  'ap-northeast-2',
+  'ap-southeast-1',
+  'ap-southeast-2',
+  'ca-central-1',
+  'eu-central-1',
+  'eu-west-1',
+  'eu-west-2',
+  'eu-west-3',
```
Comment (wchaws) [inline src/keycloak.ts:28]: why not add `cn-northwest-1` at here

File: src/keycloak.ts (hunk 1 above)
Comment (pahud) [reply src/keycloak.ts:214]: > Most of the code are the same @pahud, can we refactor it?

Hi yes, feel free to drop another PR for the refactor if you are interested.

Thanks.

Comment (pahud) [reply src/keycloak.ts:214]: @wchaws  PR ready for the refactor https://github.com/pahud/cdk-keycloak/pull/41

**Pull request merged**